src/ChannelServer/World/Character.cs

```diff
@@ -182,14 +288,17 @@ public void Move(float x, float y, float z, float dx, float dy, float unkFloat)
 		/// <param name="dy"></param>
 		public void StopMove(float x, float y, float z, float dx, float dy)
 		{
+			/*
 			this.SetPosition(x, y, z);
 			this.SetDirection(dx, dy);
 			this.IsMoving = false;
+			*/
+			this.SetDestination(new Position(x, y, z));
 
 			// Sending ZC_MOVE_STOP works as well, but it doesn't have
 			// a direction, so the character stops and looks north
 			// on other's screens.
-			Send.ZC_PC_MOVE_STOP(this, this.Position, this.Direction);
+			//Send.ZC_PC_MOVE_STOP(this, this.Position, this.Direction);
```

Comment:
If you don't send ZC_PC_MOVE_STOP here, but from the tick, the character won't stop on other clients until the tick. If the tick is several hundreds of milliseconds long, + the time it takes for the packet to arrive, you might get a big difference in what the clients display.